mantidproject/mantid
# Fix crash after script repository install location is deleted / moved

### Description of work

Fix for a crash where if you moved / deleted your local script repository install and clicked reload, mantid would crash.
Now there is a check to make sure the dir exists, and the user is prompted to reinstall if not. 
The label displaying the location is updated as well.

Sorry for slightly messy PR, I tried to do this cleverly with a QFileSystemWatcher at first.

Fixes #35579 

### To test:

- Go to `File -> Scrpit Repository` to open the interface (you will probably be prompted to install)
- Go to the file browser and delete the folder holding the script repository
- Click 'reload', should be prompted to reinstall somewhere.
- Download a script or folder using the interface.
- Go t the file browser and rename the repository folder.
- Click 'reload' and  when asked to reinstall select the renamed folder
- The file(s) you downloaded previously should still be shown as downloaded.
- Try any other combinations of things you can think of.



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
